File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
class Solution {
    private int count = 0;
    private int answer = 0;

    public int kthSmallest(TreeNode root, int k) {
        dfs(root, k);
        return answer;
    }

    private void dfs(TreeNode node, int k) {
        if (node == null) {
            return;
        }
        if (count >= k) return;

        dfs(node.left, k);

        count++;
        if (count == k) {
            answer = node.val;
            return;
        }

        dfs(node.right, k);
    }
}

File: Data Structures & Algorithms/reverse-nodes-in-k-group/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:

    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        def reverseGroup(head: Optional[ListNode], tail: Optional[ListNode]):
            prev = tail
            first = head.next
            index = 0
            while index < k:
                nextnode = first.next
                first.next = prev
                prev = first
                first = nextnode
                index += 1
            
            head.next = prev
        
        dummy = ListNode()
        dummy.next = head
        left = dummy
        right = None
        countpointer = head
        count = 0
        while countpointer != None and count < k:
            count += 1
            countpointer = countpointer.next

        while count == k:
            right = countpointer
            reverseGroup(left, right)

            for i in range(k):
                left = left.next
            countpointer = left.next
            count = 0
            while countpointer != None and count < k:
                count += 1
                countpointer = countpointer.next
        return dummy.next




